framework/common/encryption.cpp

```diff
@@ -223,33 +223,57 @@ bool hmac::digest(uint8_t *digest)
     return HMAC_Final(m_ctx, digest, &digest_length);
 }
 
-bool aes_encrypt(const uint8_t *key, const uint8_t *iv, uint8_t *data, uint32_t data_len)
+bool aes_encrypt(const uint8_t *key, const uint8_t *iv, uint8_t *plaintext, int plen,
+                 uint8_t *ciphertext, int &clen)
 {
     EVP_CIPHER_CTX *ctx;
+    int len;
 
-    int clen, len;
-    uint8_t buf[128];
+    /* Verify that the ciphertext buffer has enough storage room
+     * for block size alignment padding which will be added
+     * during encryption.
+     */
+    int padlen = plen - ((plen + 15) & 0xFU);
+    if (clen < padlen) {
```

Comment:
Changed to 
```
int padlen = plen % 16;
if (clen < plen + (padlen ? padlen : 16)) {
    LOG(ERROR) << "Insufficient room for padding in ciphertext buffer";
    return false;
}
```
Which matches PKCS#5 - if the plaintext length is a multiple of block size, the buffer for the encrypted output should be `plen + 16`, otherwise it is `plen + paddlen`.